src/exponential-backoff.js

```diff
@@ -17,10 +17,10 @@
 
 /**
  * @param {opt_base} opt_base Exponential base. Defaults to 2.
- * @return {function(function())} Function that when invoked will
+ * @return {function(function(): number)} Function that when invoked will
```

Comment:
Food for thought. Would your code be simplified if you did this:

1. Add another function here that only does the math. E.g. `getNextBackoffTimeout(baseTimeout)`
2. In the poller, have the `baseTimeout` and `currentTimeout`. Initially `currentTimeout = baseTimeout`
3. In case of failure: `currentTimeout = getNextBackoffTimeout(baseTimeout)`
4. In case of success: `currentTimeout = baseTimeout`

Thus, `isBackedOff = baseTimeout != currentTimeout`

This way you can also avoid additional jitter implementation, etc. Just to consider - if you like it used it, if not, we can finish what you have.